### PR TITLE
chore: adding shared component lib

### PR DESCRIPTION
**Pull request by siddhart1o1**

### CODE REVIEW

File: apps/consent/package.json [hunk 1]
```diff
@@ -32,13 +32,14 @@
     "edge-csrf": "^1.0.6",
     "graphql": "^16.8.1",
     "libphonenumber-js": "^1.10.51",
-    "next": "14.0.4",
+    "next": "14.0.1",
```
Comment (bodymindarts) [inline apps/consent/package.json:35]: why downgrade?

File: apps/dashboard/BUCK [hunk 2]
```diff
@@ -65,6 +65,7 @@ next_build_bin(
 
 dev_deps_srcs = {
     "lib/eslint-config": "//lib/eslint-config:src",
+    "lib/galoy-components": "//lib/galoy-components:src",
```
Comment (bodymindarts) [inline apps/dashboard/BUCK:68]: why in dev_deps?

File: toolchains/workspace-pnpm/macros.bzl [hunk 3]
```diff
@@ -494,6 +494,12 @@ _next_build = rule(
             default = "toolchains//:workspace_pnpm",
             providers = [WorkspacePnpmToolchainInfo],
         ),
+        "prod_deps_srcs": attrs.dict(
```
Comment (bodymindarts) [inline toolchains/workspace-pnpm/macros.bzl:497]: Adding this isn't going to have the desired effect. Also need to handle the argument in the implementation cc @vindard

File: apps/consent/package.json (hunk 1 above)
Comment (siddhart1o1) [reply apps/consent/package.json:35]: Cypress tests were failing with this version; they were flaky. I will look into this, but I think that would be a topic for another PR.

File: toolchains/workspace-pnpm/macros.bzl (hunk 3 above)
Comment (siddhart1o1) [reply toolchains/workspace-pnpm/macros.bzl:497]: > Adding this isn't going to have the desired effect. Also need to handle the argument in the implementation cc @vindard

https://github.com/GaloyMoney/galoy/blob/3c150ca73d2a8a81277d58d50ef5e2e128272576/toolchains/workspace-pnpm/macros.bzl#L568
I think `prod_deps_srcs`  handles this, but yes, @vindard may have more context.

File: apps/dashboard/BUCK (hunk 2 above)
Comment (siddhart1o1) [reply apps/dashboard/BUCK]: yes this should be in prod_deps.

**Review by bodymindarts — APPROVED** [review]

**Pull request merged**